File: dart/resistor-color/lib/resistor_color.dart
class ResistorColor {
  List<String> colors = [
    'black',
    'brown',
    'red',
    'orange',
    'yellow',
    'green',
    'blue',
    'violet',
    'grey',
    'white'
  ];

  int colorCode(String color) {
    int result = 0;
    for (int i = 0; i < colors.length; i++) {
      if (colors[i] == color) {
        result = i;
      }
    }

    return result;
  }
}

File: src/c/resistor-color/resistor_color.c
#include "resistor_color.h"

static resistor_band_t all_colors[10];

int color_code(resistor_band_t color) {
  return (int)color - 1;
}

resistor_band_t* colors() {
  for (resistor_band_t c = COLOR_START + 1; c < COLOR_TAIL; ++c) {
    all_colors[c - 1] = c;
  }

  return all_colors;
}
